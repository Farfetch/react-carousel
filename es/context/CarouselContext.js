import React from 'react';
export default /*#__PURE__*/React.createContext({
  goNext: function goNext() {
    return null;
  },
  goPrev: function goPrev() {
    return null;
  },
  goTo: function goTo() {
    return null;
  },
  setActiveItem: function setActiveItem() {
    return null;
  },
  setItemsLength: function setItemsLength() {
    return null;
  },
  setIsMovementBlocked: function setIsMovementBlocked() {
    return null;
  },
  activeItem: 0,
  isMovementBlocked: false,
  isRTL: false,
  itemsToShow: 0,
  itemsLength: 0,
  itemsToScroll: 0,
  direction: null
});