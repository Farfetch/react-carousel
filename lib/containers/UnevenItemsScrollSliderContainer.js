"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _context = require("../context");

var _components = require("../components");

var _react = _interopRequireWildcard(require("react"));

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var UnevenItemsScrollSliderContainer = function UnevenItemsScrollSliderContainer(props) {
  var _useContext = (0, _react.useContext)(_context.CarouselContext),
      isRTL = _useContext.isRTL,
      activeItem = _useContext.activeItem,
      itemsLength = _useContext.itemsLength,
      setItemsLength = _useContext.setItemsLength,
      goTo = _useContext.goTo,
      isInfinite = _useContext.isInfinite,
      direction = _useContext.direction;

  return /*#__PURE__*/_react["default"].createElement(_components.UnevenItemsScrollSlider, _extends({}, props, {
    isRTL: isRTL,
    activeItem: activeItem,
    itemsLength: itemsLength,
    setItemsLength: setItemsLength,
    goTo: goTo,
    isInfinite: isInfinite,
    direction: direction
  }));
};

var _default = UnevenItemsScrollSliderContainer;
exports["default"] = _default;
module.exports = exports.default;