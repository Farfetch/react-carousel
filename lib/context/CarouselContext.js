"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var _default = /*#__PURE__*/_react["default"].createContext({
  goNext: function goNext() {
    return null;
  },
  goPrev: function goPrev() {
    return null;
  },
  goTo: function goTo() {
    return null;
  },
  setActiveItem: function setActiveItem() {
    return null;
  },
  setItemsLength: function setItemsLength() {
    return null;
  },
  setIsMovementBlocked: function setIsMovementBlocked() {
    return null;
  },
  activeItem: 0,
  isMovementBlocked: false,
  isRTL: false,
  itemsToShow: 0,
  itemsLength: 0,
  itemsToScroll: 0,
  direction: null
});

exports["default"] = _default;
module.exports = exports.default;