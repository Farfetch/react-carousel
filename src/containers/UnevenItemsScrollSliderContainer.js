import { CarouselContext } from '#context';
import { UnevenItemsScrollSlider } from '#components';
import React, { useContext } from 'react';

const UnevenItemsScrollSliderContainer = (props) => {
    const {
        isRTL,
        activeItem,
        itemsLength,
        setItemsLength,
        goTo,
        isInfinite,
        direction,
    } = useContext(CarouselContext);

    return (
        <UnevenItemsScrollSlider
            {...props}
            isRTL={isRTL}
            activeItem={activeItem}
            itemsLength={itemsLength}
            setItemsLength={setItemsLength}
            goTo={goTo}
            isInfinite={isInfinite}
            direction={direction}
        />
    );
};

export default UnevenItemsScrollSliderContainer;
