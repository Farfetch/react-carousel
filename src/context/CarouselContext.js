import React from 'react';

export default React.createContext({
    goNext: () => null,
    goPrev: () => null,
    goTo: () => null,
    setActiveItem: () => null,
    setItemsLength: () => null,
    setIsMovementBlocked: () => null,
    activeItem: 0,
    isMovementBlocked: false,
    isRTL: false,
    itemsToShow: 0,
    itemsLength: 0,
    itemsToScroll: 0,
    direction: null,
});
