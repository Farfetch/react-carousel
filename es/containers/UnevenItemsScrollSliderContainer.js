function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { CarouselContext } from "../context";
import { UnevenItemsScrollSlider } from "../components";
import React, { useContext } from 'react';

var UnevenItemsScrollSliderContainer = function UnevenItemsScrollSliderContainer(props) {
  var _useContext = useContext(CarouselContext),
      isRTL = _useContext.isRTL,
      activeItem = _useContext.activeItem,
      itemsLength = _useContext.itemsLength,
      setItemsLength = _useContext.setItemsLength,
      goTo = _useContext.goTo,
      isInfinite = _useContext.isInfinite,
      direction = _useContext.direction;

  return /*#__PURE__*/React.createElement(UnevenItemsScrollSlider, _extends({}, props, {
    isRTL: isRTL,
    activeItem: activeItem,
    itemsLength: itemsLength,
    setItemsLength: setItemsLength,
    goTo: goTo,
    isInfinite: isInfinite,
    direction: direction
  }));
};

export default UnevenItemsScrollSliderContainer;