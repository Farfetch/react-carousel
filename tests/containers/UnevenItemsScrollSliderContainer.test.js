import { createSerializer } from 'enzyme-to-json';
// At time of writing useContext does not work with enzyme for shallow rendering
import { mount } from 'enzyme';
import CarouselContext from '../../src/context/CarouselContext';
import React from 'react';
import UnevenItemsScrollSliderContainer from '../../src/containers/UnevenItemsScrollSliderContainer';

const mockContextCarousel = {
    activeItem: 1,
    isRTL: false,
    itemsLength: 1,
    setItemsLength: jest.fn(),
    goTo: jest.fn(),
};

const mockChildren = <div />;

beforeAll(() => {
    expect.addSnapshotSerializer(createSerializer());
});

describe('<UnevenItemsScrollSliderContainer/>', () => {
    it('should render correctly', () => {
        const tree = mount(
            <CarouselContext.Provider value={mockContextCarousel}>
                <UnevenItemsScrollSliderContainer>
                    {mockChildren}
                </UnevenItemsScrollSliderContainer>
            </CarouselContext.Provider>
        );

        expect(tree).toMatchSnapshot();
    });

    it('should render correctly with custom Props', () => {
        const mockTheme = {
            container: 'container',
        };
        const tree = mount(
            <CarouselContext.Provider value={mockContextCarousel}>
                <UnevenItemsScrollSliderContainer
                    data-testId="test"
                    theme={mockTheme}
                >
                    {mockChildren}
                </UnevenItemsScrollSliderContainer>
            </CarouselContext.Provider>
        );

        expect(
            tree.find('BaseUnevenItemsScrollSlider').prop('data-testId')
        ).toBe('test');
        expect(tree.find('BaseUnevenItemsScrollSlider').prop('theme')).toEqual(
            mockTheme
        );
    });

    it('should render correctly with correct Context', () => {
        const tree = mount(
            <CarouselContext.Provider value={mockContextCarousel}>
                <UnevenItemsScrollSliderContainer>
                    {mockChildren}
                </UnevenItemsScrollSliderContainer>
            </CarouselContext.Provider>
        );

        expect(tree.find('BaseUnevenItemsScrollSlider').props()).toMatchObject(
            mockContextCarousel
        );
    });
});
