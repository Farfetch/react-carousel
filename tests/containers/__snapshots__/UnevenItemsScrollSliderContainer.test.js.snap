// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UnevenItemsScrollSliderContainer/> should render correctly 1`] = `
<UnevenItemsScrollSliderContainer>
  <BaseUnevenItemsScrollSlider
    activeItem={1}
    goTo={[MockFunction]}
    isRTL={false}
    itemsLength={1}
    ratioToScroll={1}
    setItemsLength={
      [MockFunction] {
        "calls": Array [
          Array [
            NaN,
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
  >
    <div
      className="unevenItemsContainer"
      onScroll={[Function]}
    >
      <div
        data-element="unevenItemsWrapper"
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
      >
        <div
          key="null/.0"
          style={Object {}}
        />
      </div>
    </div>
  </BaseUnevenItemsScrollSlider>
</UnevenItemsScrollSliderContainer>
`;
